hudi-common/src/main/java/org/apache/hudi/common/model/DefaultHoodieRecordPayload.java

```diff
@@ -86,30 +86,27 @@ public Option<IndexedRecord> getInsertValue(Schema schema, Properties properties
     GenericRecord incomingRecord = HoodieAvroUtils.bytesToAvro(recordBytes, schema);
     eventTime = updateEventTime(incomingRecord, properties);
 
-    return isDeleteRecord(incomingRecord, properties) ? Option.empty() : Option.of(incomingRecord);
+    return isDeleted(schema, properties) ? Option.empty() : Option.of(incomingRecord);
```

Comment:
is this incomingRecord or schema? intentional change?

Reply:
is this to avoid double parsing of deleted record?

Reply:
Yes, this is to avoid double parsing.  After the instantiation, whether the record is a delete record is already determined, stored in `isDeletedRecord`, and `isDeleted(schema, properties)` returns the result.
```
  /**
   * Defines whether this implementation of {@link HoodieRecordPayload} is deleted.
   * We will not do deserialization in this method.
   */
  public boolean isDeleted(Schema schema, Properties props) {
    return isDeletedRecord;
  }
```